lib/sensu-handler.rb

```diff
@@ -42,28 +42,13 @@ def filter
       end
     end
 
-    # Evaluates whether deprecated filtering is explicitly disabled
-    # via global Sensu configuration. Defaults to false,
-    # i.e. deprecated filters are run by default.
-    #
-    # @return [TrueClass, FalseClass]
-    def deprecated_filtering_globally_disabled?
-      global_settings = settings.fetch('sensu_plugin', {})
-      global_settings['disable_deprecated_filtering'].to_s == "true"
-    end
-
     # Evaluates whether the event should be processed by any of the
     # filter methods in this library. Defaults to true,
     # i.e. deprecated filters are run by default.
     #
     # @return [TrueClass, FalseClass]
     def deprecated_filtering_enabled?
-      if deprecated_filtering_globally_disabled?
-        return false
-      else
-        @event['check']['enable_deprecated_filtering'].nil? || \
-        @event['check']['enable_deprecated_filtering'].to_s == "true"
-      end
+      @event['check']['enable_deprecated_filtering'].to_s == "true"
```

Comment:
@portertech I believe the `nil` is exercised in the`test_occurrence_filter_deprecation_warning_not_present_by_default` unit test. Locally I've explicitly added `"enable_deprecated_filtering" => nil` to the event.check scope, the test outcome is the same.

Would you prefer the use of `#fetch` ?